app/main.py

```diff
@@ -1,52 +1,16 @@
-import json
-import xml.etree.ElementTree as ET
+from __future__ import annotations
 
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.entities.book import Book
+from app.dispatchers.command_dispatcher import CommandDispatcher
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
```

Comment:
To follow the SRP more strictly you can consider returning only one type in main `str` f.e.

Reply:
@Oleg-s-s  but I don't know how handle this in better way. Because I have 2 options, some `cmd` should be printed some returned, so this function always should be None | str

Reply:
or you mean avoid logic when something only printed, and use annotation just for returning cmd